core/src/avm2/object/class_object.rs

```diff
@@ -69,10 +71,19 @@ pub struct ClassObjectData<'gc> {
     /// as `None` here. AVM2 considers both applications to be separate
     /// classes, though we consider the parameter to be the class `Object` when
     /// we get a param of `null`.
-    applications: HashMap<Option<ClassObject<'gc>>, ClassObject<'gc>>,
+    applications: FnvHashMap<Option<ClassObject<'gc>>, ClassObject<'gc>>,
 
     /// Interfaces implemented by this class.
     interfaces: Vec<ClassObject<'gc>>,
+
+    /// All traits present on instances of this class.
+    ///
+    /// This is a comprehensive list of all traits, including all superclass
+    /// and interface traits, as well as the superclass that defined the trait.
+    resolved_instance_traits: PropertyMap<'gc, Vec<(ClassObject<'gc>, Trait<'gc>)>>,
+
+    /// All traits present on the class itself.
+    resolved_class_traits: PropertyMap<'gc, Vec<Trait<'gc>>>,
```

Comment:
kinda surprised why this doesn't need reference to ClassObject, like the one above.

Reply:
The reason why I didn't include the `ClassObject` here is because class traits aren't inherited. You can access the superclass's class traits as scope variables because the superclass becomes part of the outerscope for all class method traits and the constructor.